### ISSUE TITLE
ENH: errors='ignore' should work for record_path for pandas.json_normalize function

### ISSUE THREAD

**Issue by averhagen** [pandas-dev/pandas#44329]
When trying to normalize the example JSON data with pandas.json_normalize I get a KeyError exception even when I set errors='ignore'.

pandas-version=1.3.4
python-version=3.7.5

python code used:

```
import json
import pandas as pd

file_path = Path('path/to/example/json-data.json')
json_data = json.loads(file_path.read_text())
pd.json_normalize(json_data, record_path='Sales', errors='ignore')
```

`[
	{
		"TransactionDate": "2019-12-01T19:58:13",
		"PurchaseLocation": "Web",
		"ReceiptNo": 1,
		"PostalState": "KY"
	},
	{
		"TransactionDate": "2019-12-01T12:51:47",
		"PurchaseLocation": "POS",
		"ReceiptNo": 2,
		"PostalState": "KY",
		"Sales": [
			{
				"ProductID": 101024,
				"Quantity": 1
			}
		]
	}
]`

I expected that this should not throw a KeyError and should return a dataframe just without the first JSON object. Having re-looked at the documentation, it looks like the "errors" parameter only applys to the "meta" values. I would find it very useful if we could specify the function to ignore KeyErrors related to the record path

**Comment by rhshadrach** [comment]
Can you provide the contents of the json file.

**Comment by averhagen** [comment]
@rhshadrach 
Yes, I can. The contents are actually already in the comment too.

`[ { "TransactionDate": "2019-12-01T19:58:13", "PurchaseLocation": "Web", "ReceiptNo": 1, "PostalState": "KY" }, { "TransactionDate": "2019-12-01T12:51:47", "PurchaseLocation": "POS", "ReceiptNo": 2, "PostalState": "KY", "Sales": [ { "ProductID": 101024, "Quantity": 1 } ] } ]`

**Comment by rhshadrach** [comment]
Ah, I missed that, thanks. You're most likely already aware, but you can accomplish the desired behavior via

    pd.json_normalize([e for e in json_data if 'Sales' in e], record_path='Sales', errors='ignore')

I don't use json, so I don't have any opinion on whether errors should apply in this case or not.

**Comment by SamiEldaw** [comment]
@rhshadrach I get "TypeError: string indices must be integers" when executing your reply in a similar situation

**Comment by rhshadrach** [comment]
@SamiEldaw - on what version of pandas? I just reran on main and did not get any error.

**Comment by SamiEldaw** [comment]
@rhshadrach  I am on 1.4.4, here is my json: 

{'Packages': ['Economics'],
 'PSharpScripts': ['Economics Ownership Determination',
  'Economics Ownership Income And Cash Flow',
  'Economics Ownership Operating Expenses and Production Taxes',
  'Economics Ownership Sales Volumes And Revenue'],
 'Signal': ['working interest with reversion']}

I loop through a list and generate urls and execute api calls, each call will have a different Json structure, each loop I append the normalized json to a data frame using the same call and record path, hence the need to ignore record path not found error, however when executing the below I get the type error

PsharpScripts_Outt= PsharpScripts_Outt.append(pd.json_normalize([r for r in petr8 if "PSharpScripts" in r] , record_path=['PSharpScripts'],meta=['Signal','workflow', 'Packages'], errors='ignore'))

petr8 is my json response, in some iterations it will not have the key PSharpScripts
